diff --git a/build/config/compiler/BUILD.gn b/build/config/compiler/BUILD.gn
index d0b55ce295fdf..7411a76d4e87c 100644
--- a/build/config/compiler/BUILD.gn
+++ b/build/config/compiler/BUILD.gn
@@ -349,6 +349,7 @@ config("compiler") {
 
     if (use_lld) {
       ldflags += [ "-fuse-ld=lld" ]
+      ldflags += [ "-Wl,--threads=8" ]
       if (lld_path != "") {
         ldflags += [ "-B$lld_path" ]
       }
@@ -666,11 +667,28 @@ config("compiler") {
     cflags += [
       "-flto=thin",
       "-fsplit-lto-unit",
+      "-mllvm", "-adce-remove-loops",
+      "-mllvm", "-aggressive-ext-opt",
+      "-mllvm", "-enable-cse-in-irtranslator",
+      "-mllvm", "-enable-cse-in-legalizer",
+      "-mllvm", "-enable-gvn-hoist",
+      "-mllvm", "-gvn-hoist-max-bbs=-1",
+      "-mllvm", "-gvn-hoist-max-depth=-1",
+      "-mllvm", "-gvn-hoist-max-chain-length=-1",
+      "-mllvm", "-enable-interleaved-mem-accesses",
+      "-mllvm", "-enable-masked-interleaved-mem-accesses",
+      "-mllvm", "-enable-loopinterchange",
+      "-mllvm", "-enable-loop-distribute",
+      "-mllvm", "-enable-loop-flatten",
+      "-mllvm", "-interleave-small-loop-scalar-reduction",
+      "-mllvm", "-loop-rotate-multi",
+      "-mllvm", "-scalar-evolution-use-expensive-range-sharpening",
+      "-mllvm", "-extra-vectorizer-passes",
     ]
 
     # Limit the size of the ThinLTO cache to the lesser of 10% of
     # available disk space, 40GB and 100000 files.
-    cache_policy = "cache_size=10%:cache_size_bytes=40g:cache_size_files=100000"
+    # cache_policy = "cache_size=10%:cache_size_bytes=40g:cache_size_files=100000"
 
     # An import limit of 30 has better performance (per speedometer) and lower
     # binary size than the default setting of 100.
@@ -682,12 +700,66 @@ config("compiler") {
       ldflags += [
         "/opt:lldltojobs=all",
         "-mllvm:-import-instr-limit=$import_instr_limit",
-        "/lldltocache:" +
-            rebase_path("$root_out_dir/thinlto-cache", root_build_dir),
-        "/lldltocachepolicy:$cache_policy",
+        # "/lldltocache:" +
+        #     rebase_path("$root_out_dir/thinlto-cache", root_build_dir),
+        # "/lldltocachepolicy:$cache_policy",
+        "-mllvm:-adce-remove-loops",
+        "-mllvm:-aggressive-ext-opt",
+        "-mllvm:-enable-cse-in-irtranslator",
+        "-mllvm:-enable-cse-in-legalizer",
+        "-mllvm:-enable-gvn-hoist",
+        "-mllvm:-gvn-hoist-max-bbs=-1",
+        "-mllvm:-gvn-hoist-max-depth=-1",
+        "-mllvm:-gvn-hoist-max-chain-length=-1",
+        "-mllvm:-enable-interleaved-mem-accesses",
+        "-mllvm:-enable-masked-interleaved-mem-accesses",
+        "-mllvm:-enable-loopinterchange",
+        "-mllvm:-enable-loop-distribute",
+        "-mllvm:-enable-loop-flatten",
+        "-mllvm:-interleave-small-loop-scalar-reduction",
+        "-mllvm:-loop-rotate-multi",
+        "-mllvm:-scalar-evolution-use-expensive-range-sharpening",
+        "-mllvm:-extra-vectorizer-passes",
+        "-mllvm:-polly",
+        "-mllvm:-polly-2nd-level-tiling",
+        "-mllvm:-polly-detect-profitability-min-per-loop-insts=40",
+        "-mllvm:-polly-invariant-load-hoisting",
+        "-mllvm:-polly-loopfusion-greedy",
+        "-mllvm:-polly-register-tiling",
+        "-mllvm:-polly-run-dce",
+        "-mllvm:-polly-vectorizer=stripmine",
+        # "-mllvm:-pass-remarks-analysis=polly",
       ]
     } else {
-      ldflags += [ "-flto=thin" ]
+      ldflags += [
+        "-flto=thin",
+        "-Wl,-mllvm,-adce-remove-loops",
+        "-Wl,-mllvm,-aggressive-ext-opt",
+        "-Wl,-mllvm,-enable-cse-in-irtranslator",
+        "-Wl,-mllvm,-enable-cse-in-legalizer",
+        "-Wl,-mllvm,-enable-gvn-hoist",
+        "-Wl,-mllvm,-gvn-hoist-max-bbs=-1",
+        "-Wl,-mllvm,-gvn-hoist-max-depth=-1",
+        "-Wl,-mllvm,-gvn-hoist-max-chain-length=-1",
+        "-Wl,-mllvm,-enable-interleaved-mem-accesses",
+        "-Wl,-mllvm,-enable-masked-interleaved-mem-accesses",
+        "-Wl,-mllvm,-enable-loopinterchange",
+        "-Wl,-mllvm,-enable-loop-distribute",
+        "-Wl,-mllvm,-enable-loop-flatten",
+        "-Wl,-mllvm,-interleave-small-loop-scalar-reduction",
+        "-Wl,-mllvm,-loop-rotate-multi",
+        "-Wl,-mllvm,-scalar-evolution-use-expensive-range-sharpening",
+        "-Wl,-mllvm,-extra-vectorizer-passes",
+        "-Wl,-mllvm,-polly",
+        "-Wl,-mllvm,-polly-2nd-level-tiling",
+        "-Wl,-mllvm,-polly-detect-profitability-min-per-loop-insts=40",
+        "-Wl,-mllvm,-polly-invariant-load-hoisting",
+        "-Wl,-mllvm,-polly-loopfusion-greedy",
+        "-Wl,-mllvm,-polly-register-tiling",
+        "-Wl,-mllvm,-polly-run-dce",
+        "-Wl,-mllvm,-polly-vectorizer=stripmine",
+        # "-Wl,-mllvm,-pass-remarks-analysis=polly",
+      ]
 
       # Enabling ThinLTO on Chrome OS too, in an effort to reduce the memory
       # usage in crbug.com/1038040. Note this will increase build time in
@@ -705,12 +777,12 @@ config("compiler") {
           "-Wcrl,object_path_lto",
         ]
       } else {
-        ldflags +=
-            [ "-Wl,--thinlto-cache-dir=" +
-              rebase_path("$root_out_dir/thinlto-cache", root_build_dir) ]
+        # ldflags +=
+        #     [ "-Wl,--thinlto-cache-dir=" +
+        #       rebase_path("$root_out_dir/thinlto-cache", root_build_dir) ]
       }
 
-      ldflags += [ "-Wl,--thinlto-cache-policy=$cache_policy" ]
+      # ldflags += [ "-Wl,--thinlto-cache-policy=$cache_policy" ]
 
       if (is_chromeos) {
         # ARM was originally set lower than x86 to keep the size
@@ -886,6 +958,8 @@ config("compiler") {
     # For deterministic builds, keep the local machine's current working
     # directory from appearing in build outputs.
     "-Zremap-cwd-prefix=.",
+
+    "-Ctarget-feature=+avx,+aes",
   ]
   if (rust_abi_target != "") {
     rustflags += [ "--target=$rust_abi_target" ]
@@ -918,12 +992,14 @@ config("rust_defer_lto_to_linker") {
 # building with ThinLTO, no optimization is performed in the link step.
 config("thinlto_optimize_default") {
   if (!is_debug && use_thin_lto && is_a_target_toolchain) {
-    lto_opt_level = 0
+    lto_opt_level = 3
 
     if (is_win) {
       ldflags = [ "/opt:lldlto=" + lto_opt_level ]
+      ldflags += [ "-opt:lldltocgo=" + lto_opt_level ]
     } else {
       ldflags = [ "-Wl,--lto-O" + lto_opt_level ]
+      ldflags += [ "-Wl,--lto-CGO" + lto_opt_level ]
     }
 
     rustflags = [ "-Clto=thin" ]
@@ -941,15 +1017,17 @@ config("thinlto_optimize_default") {
 config("thinlto_optimize_max") {
   if (!is_debug && use_thin_lto && is_a_target_toolchain) {
     if (thin_lto_enable_optimizations) {
-      lto_opt_level = 2
+      lto_opt_level = 3
     } else {
-      lto_opt_level = 0
+      lto_opt_level = 3
     }
 
     if (is_win) {
       ldflags = [ "/opt:lldlto=" + lto_opt_level ]
+      ldflags += [ "-opt:lldltocgo=" + lto_opt_level ]
     } else {
       ldflags = [ "-Wl,--lto-O" + lto_opt_level ]
+      ldflags += [ "-Wl,--lto-CGO" + lto_opt_level ]
     }
 
     rustflags = [ "-Clto=thin" ]
@@ -983,7 +1061,7 @@ config("compiler_cpu_abi") {
     if (current_cpu == "x64") {
       cflags += [
         "-m64",
-        "-msse3",
+        "-march=native",
       ]
       ldflags += [ "-m64" ]
     } else if (current_cpu == "x86") {
@@ -2063,15 +2141,16 @@ if (is_win) {
     # Aix does not support these.
 
     common_optimize_on_cflags += [
-      # Put data and code in their own sections, so that unused symbols
+      # P ut data and code in their own sections, so that unused symbols
       # can be removed at link time with --gc-sections.
       "-fdata-sections",
       "-ffunction-sections",
+      "-funique-section-names",
     ]
     if ((!is_nacl || is_nacl_saigo) && is_clang) {
       # We don't care about unique section names, this makes object files a bit
       # smaller.
-      common_optimize_on_cflags += [ "-fno-unique-section-names" ]
+      # common_optimize_on_cflags += [ "-fno-unique-section-names" ]
     }
 
     common_optimize_on_ldflags += [
@@ -2116,12 +2195,12 @@ config("optimize") {
     if (chrome_pgo_phase != 2) {
       # Favor size over speed, /O1 must be before the common flags.
       # /O1 implies /Os and /GF.
-      cflags = [ "/O1" ] + common_optimize_on_cflags + [ "/Oi" ]
+      cflags = [ "-Xclang", "-O3", ] + common_optimize_on_cflags + [ "/Oi" ]
       rustflags = [ "-Copt-level=s" ]
     } else {
       # PGO requires all translation units to be compiled with /O2. The actual
       # optimization level will be decided based on the profiling data.
-      cflags = [ "/O2" ] + common_optimize_on_cflags + [ "/Oi" ]
+      cflags = [ "-Xclang", "-O3", ] + common_optimize_on_cflags + [ "/Oi" ]
 
       # https://doc.rust-lang.org/rustc/profile-guided-optimization.html#usage
       # suggests not using an explicit `-Copt-level` at all, and the default is
@@ -2131,7 +2210,7 @@ config("optimize") {
   } else if (optimize_for_size) {
     # Favor size over speed.
     if (is_clang) {
-      cflags = [ "-Oz" ] + common_optimize_on_cflags
+      cflags = [ "-O3" ] + common_optimize_on_cflags
 
       if (use_ml_inliner && is_a_target_toolchain) {
         cflags += [
@@ -2140,12 +2219,12 @@ config("optimize") {
         ]
       }
     } else {
-      cflags = [ "-Os" ] + common_optimize_on_cflags
+      cflags = [ "-O3" ] + common_optimize_on_cflags
     }
 
     # Like with `-Oz` on Clang, `-Copt-level=z` will also turn off loop
     # vectorization.
-    rustflags = [ "-Copt-level=z" ]
+    rustflags = [ "-Copt-level=s" ]
   } else if (is_chromeos) {
     # TODO(gbiv): This is partially favoring size over speed. CrOS exclusively
     # uses clang, and -Os in clang is more of a size-conscious -O2 than "size at
@@ -2159,7 +2238,7 @@ config("optimize") {
     # vectorization while otherwise optimizing for size.
     rustflags = [ "-Copt-level=s" ]
   } else {
-    cflags = [ "-O2" ] + common_optimize_on_cflags
+    cflags = [ "-Xclang", "-O3", ] + common_optimize_on_cflags
 
     # The `-O3` for clang turns on extra optimizations compared to the standard
     # `-O2`. But for rust, `-Copt-level=3` is the default and is thus reliable
@@ -2226,11 +2305,11 @@ config("optimize_max") {
     if (is_win) {
       # Favor speed over size, /O2 must be before the common flags.
       # /O2 implies /Ot, /Oi, and /GF.
-      cflags = [ "/O2" ] + common_optimize_on_cflags
+      cflags = [ "-Xclang", "-O3", ] + common_optimize_on_cflags
     } else if (optimize_for_fuzzing) {
       cflags = [ "-O1" ] + common_optimize_on_cflags
     } else {
-      cflags = [ "-O2" ] + common_optimize_on_cflags
+      cflags = [ "-O3" ] + common_optimize_on_cflags
     }
     rustflags = [ "-Copt-level=3" ]
   }
@@ -2259,7 +2338,33 @@ config("optimize_speed") {
     if (is_win) {
       # Favor speed over size, /O2 must be before the common flags.
       # /O2 implies /Ot, /Oi, and /GF.
-      cflags = [ "/O2" ] + common_optimize_on_cflags
+      cflags = [ "-Xclang", "-O3", ] + common_optimize_on_cflags
+    } else if (optimize_for_fuzzing) {
+      cflags = [ "-O1" ] + common_optimize_on_cflags
+    } else {
+      cflags = [ "-O3", ] + common_optimize_on_cflags
+    }
+    rustflags = [ "-Copt-level=3" ]
+  }
+}
+
+config("optimize_v8") {
+  if (is_nacl && is_nacl_irt) {
+    # The NaCl IRT is a special case and always wants its own config.
+    # Various components do:
+    #   if (!is_debug) {
+    #     configs -= [ "//build/config/compiler:default_optimization" ]
+    #     configs += [ "//build/config/compiler:optimize_max" ]
+    #   }
+    # So this config has to have the selection logic just like
+    # "default_optimization", below.
+    configs = [ "//build/config/nacl:irt_optimize" ]
+  } else {
+    ldflags = common_optimize_on_ldflags
+    if (is_win) {
+      # Favor speed over size, /O2 must be before the common flags.
+      # /O2 implies /Ot, /Oi, and /GF.
+      cflags = [ "-Xclang", "-O3", ] + common_optimize_on_cflags
     } else if (optimize_for_fuzzing) {
       cflags = [ "-O1" ] + common_optimize_on_cflags
     } else {
@@ -2610,7 +2715,7 @@ config("minimal_symbols") {
 # names in the final debug information.
 config("no_symbols") {
   if (is_win) {
-    ldflags = [ "/DEBUG" ]
+    ldflags = [ "/DEBUG:NONE" ]
 
     # All configs using /DEBUG should include this:
     configs = [ ":win_pdbaltpath" ]
diff --git a/build/config/win/BUILD.gn b/build/config/win/BUILD.gn
index 507a76edd4439..dff50590b5725 100644
--- a/build/config/win/BUILD.gn
+++ b/build/config/win/BUILD.gn
@@ -88,7 +88,7 @@ config("compiler") {
 
   cflags += [
     # Work around crbug.com/526851, bug in VS 2015 RTM compiler.
-    "/Zc:sizedDealloc-",
+    # "/Zc:sizedDealloc-",
   ]
 
   if (is_clang) {
@@ -129,7 +129,7 @@ config("compiler") {
     # microarchitecture. MSVC only supports a subset of architectures, and the
     # next step after SSE2 will be AVX.
     if (current_cpu == "x86" || current_cpu == "x64") {
-      cflags += [ "-msse3" ]
+      cflags += [ "-mavx", "-maes", ]
     }
 
     # Enable ANSI escape codes if something emulating them is around (cmd.exe
@@ -192,9 +192,8 @@ config("compiler") {
     ]
 
     if (use_lld) {
-      # String tail merging leads to smaller binaries, but they don't compress
-      # as well, leading to increased mini_installer size (crbug.com/838449).
-      ldflags += [ "/OPT:NOLLDTAILMERGE" ]
+      ldflags += [ "/OPT:LLDTAILMERGE" ]
+      ldflags += [ "/threads:8" ]
     }
 
     # TODO(siggi): Is this of any use anymore?
@@ -202,7 +201,7 @@ config("compiler") {
     # PDB file by about 5%) but does not otherwise alter the output binary. It
     # is enabled opportunistically for builds where it is not prohibited (not
     # supported when incrementally linking, or using /debug:fastlink).
-    ldflags += [ "/PROFILE" ]
+    # ldflags += [ "/PROFILE" ]
   }
 
   # arflags apply only to static_libraries. The normal linker configs are only
diff --git a/build/toolchain/win/win_toolchain_data.gni b/build/toolchain/win/win_toolchain_data.gni
index 505d0ce5049f1..00fcf1172c017 100644
--- a/build/toolchain/win/win_toolchain_data.gni
+++ b/build/toolchain/win/win_toolchain_data.gni
@@ -28,16 +28,4 @@ declare_args() {
                     "environment.x64",
                   ],
                   "scope")
-
-  win_toolchain_data_arm64 =
-      exec_script("//build/toolchain/win/setup_toolchain.py",
-                  [
-                    visual_studio_path,
-                    windows_sdk_path,
-                    visual_studio_runtime_dirs,
-                    "win",
-                    "arm64",
-                    "environment.arm64",
-                  ],
-                  "scope")
 }
diff --git a/media/base/supported_types.cc b/media/base/supported_types.cc
index 07b991932aad5..75a2e0eabe55d 100644
--- a/media/base/supported_types.cc
+++ b/media/base/supported_types.cc
@@ -209,33 +209,7 @@ bool IsAudioCodecProprietary(AudioCodec codec) {
 #endif  // !BUILDFLAG(USE_PROPRIETARY_CODECS)
 
 bool IsHevcProfileSupported(const VideoType& type) {
-  if (!IsColorSpaceSupported(type.color_space))
-    return false;
-
-#if BUILDFLAG(ENABLE_PLATFORM_HEVC)
-#if BUILDFLAG(PLATFORM_HAS_OPTIONAL_HEVC_SUPPORT)
-#if BUILDFLAG(IS_CHROMEOS_LACROS)
-  // TODO(b/171813538): For Lacros, the supplemental profile cache will be
-  // asking lacros-gpu, but we will be doing decoding in ash-gpu. Until the
-  // codec detection is plumbed through to ash-gpu we can do this extra check
-  // for HEVC support.
-  if (base::CommandLine::ForCurrentProcess()->HasSwitch(
-          switches::kLacrosEnablePlatformHevc)) {
-    return true;
-  }
-#endif  // BUILDFLAG(IS_CHROMEOS_LACROS)
-#if BUILDFLAG(IS_CHROMEOS_ASH)
-  if (!base::FeatureList::IsEnabled(kPlatformHEVCDecoderSupport)) {
-    return false;
-  }
-#endif  // BUILDFLAG(IS_CHROMEOS_ASH)
-  return GetSupplementalProfileCache()->IsProfileSupported(type.profile);
-#else
-  return true;
-#endif  // BUIDFLAG(PLATFORM_HAS_OPTIONAL_HEVC_SUPPORT)
-#else
-  return false;
-#endif  // BUILDFLAG(ENABLE_PLATFORM_HEVC)
+   return true;
 }
 
 bool IsVp9ProfileSupported(const VideoType& type) {
@@ -410,7 +384,7 @@ bool IsBuiltInVideoCodec(VideoCodec codec) {
   if (codec == VideoCodec::kVP8)
     return true;
 #if BUILDFLAG(USE_PROPRIETARY_CODECS)
-  if (codec == VideoCodec::kH264)
+  if (codec == VideoCodec::kH264 || codec == VideoCodec::kHEVC)
     return true;
 #endif  // BUILDFLAG(USE_PROPRIETARY_CODECS)
 #endif  // BUILDFLAG(ENABLE_FFMPEG_VIDEO_DECODERS)
diff --git a/media/filters/ffmpeg_video_decoder.cc b/media/filters/ffmpeg_video_decoder.cc
index 6b5e48095e4bc..3701bdfef832a 100644
--- a/media/filters/ffmpeg_video_decoder.cc
+++ b/media/filters/ffmpeg_video_decoder.cc
@@ -74,7 +74,6 @@ static int GetFFmpegVideoDecoderThreadCount(const VideoDecoderConfig& config) {
     case VideoCodec::kUnknown:
     case VideoCodec::kVC1:
     case VideoCodec::kMPEG2:
-    case VideoCodec::kHEVC:
     case VideoCodec::kVP9:
     case VideoCodec::kAV1:
     case VideoCodec::kDolbyVision:
@@ -86,6 +85,7 @@ static int GetFFmpegVideoDecoderThreadCount(const VideoDecoderConfig& config) {
       // No extra threads for these codecs.
       break;
 
+    case VideoCodec::kHEVC:
     case VideoCodec::kH264:
     case VideoCodec::kVP8:
       // Normalize to three threads for 1080p content, then scale linearly
