diff --git a/BUILD.gn b/BUILD.gn
index fd8ba56a01..2222a6d03d 100644
--- a/BUILD.gn
+++ b/BUILD.gn
@@ -1064,7 +1064,7 @@ config("toolchain") {
   defines = []
   cflags = []
   ldflags = []
-
+  cflags += [ "-mllvm", "-enable-ext-tsp-block-placement=false", ]
   if (v8_current_cpu == "arm") {
     defines += [ "V8_TARGET_ARCH_ARM" ]
     if (arm_version >= 7) {
@@ -1502,9 +1502,9 @@ config("always_turbofanimize") {
   # TODO(crbug.com/621335) Rework this so that we don't have the confusion
   # between "optimize_speed" and "optimize_max".
   if (((is_posix && !is_android) || is_fuchsia) && !using_sanitizer) {
-    configs += [ "//build/config/compiler:optimize_speed" ]
+    configs += [ "//build/config/compiler:optimize_fast" ]
   } else {
-    configs += [ "//build/config/compiler:optimize_max" ]
+    configs += [ "//build/config/compiler:optimize_fast" ]
   }
 }
 
