diff --git a/build/config/compiler/BUILD.gn b/build/config/compiler/BUILD.gn
index bf8269f00ecf8..72f0c7b4b72b4 100644
--- a/build/config/compiler/BUILD.gn
+++ b/build/config/compiler/BUILD.gn
@@ -543,7 +543,7 @@ config("compiler") {
 
     # TODO(crbug.com/1235145): Investigate why/if this should be needed.
     if (is_win) {
-      cflags += [ "/clang:-ffp-contract=off" ]
+      # cflags += [ "/clang:-ffp-contract=off" ]
     } else {
       cflags += [ "-ffp-contract=off" ]
     }
@@ -638,9 +638,31 @@ config("compiler") {
       "-fsplit-lto-unit",
     ]
 
+    cflags += [
+        "-mllvm", "-adce-remove-loops",
+        "-mllvm", "-enable-cse-in-irtranslator",
+        "-mllvm", "-enable-cse-in-legalizer",
+        "-mllvm", "-scalar-evolution-use-expensive-range-sharpening",
+        "-mllvm", "-loop-rotate-multi",
+        "-mllvm", "-enable-loopinterchange",
+        "-mllvm", "-enable-loop-distribute",
+        "-mllvm", "-enable-loop-flatten",
+        "-mllvm", "-enable-loop-versioning-licm",
+        "-mllvm", "-interleave-small-loop-scalar-reduction",
+        "-mllvm", "-enable-interleaved-mem-accesses",
+        "-mllvm", "-enable-masked-interleaved-mem-accesses",
+        "-mllvm", "-enable-gvn-hoist",
+        "-mllvm", "-gvn-hoist-max-bbs=-1",
+        "-mllvm", "-gvn-hoist-max-depth=-1",
+        "-mllvm", "-gvn-hoist-max-chain-length=-1",
+        "-mllvm", "-enable-nontrivial-unswitch",
+        "-mllvm", "-enable-ext-tsp-block-placement",
+        "-mllvm", "-enable-constraint-elimination",
+    ]
+
     # Limit the size of the ThinLTO cache to the lesser of 10% of
     # available disk space, 40GB and 100000 files.
-    cache_policy = "cache_size=10%:cache_size_bytes=40g:cache_size_files=100000"
+    # cache_policy = "cache_size=10%:cache_size_bytes=40g:cache_size_files=100000"
 
     # An import limit of 30 has better performance (per speedometer) and lower
     # binary size than the default setting of 100.
@@ -652,13 +674,58 @@ config("compiler") {
       ldflags += [
         "/opt:lldltojobs=all",
         "-mllvm:-import-instr-limit=$import_instr_limit",
-        "/lldltocache:" +
-            rebase_path("$root_out_dir/thinlto-cache", root_build_dir),
-        "/lldltocachepolicy:$cache_policy",
+        # "/lldltocache:" +
+        #     rebase_path("$root_out_dir/thinlto-cache", root_build_dir),
+        # "/lldltocachepolicy:$cache_policy",
+      ]
+
+      ldflags += [
+        "-mllvm:-adce-remove-loops",
+        "-mllvm:-enable-cse-in-irtranslator",
+        "-mllvm:-enable-cse-in-legalizer",
+        "-mllvm:-scalar-evolution-use-expensive-range-sharpening",
+        "-mllvm:-loop-rotate-multi",
+        "-mllvm:-enable-loopinterchange",
+        "-mllvm:-enable-loop-distribute",
+        "-mllvm:-enable-loop-flatten",
+        "-mllvm:-enable-loop-versioning-licm",
+        "-mllvm:-interleave-small-loop-scalar-reduction",
+        "-mllvm:-enable-interleaved-mem-accesses",
+        "-mllvm:-enable-masked-interleaved-mem-accesses",
+        "-mllvm:-enable-gvn-hoist",
+        "-mllvm:-gvn-hoist-max-bbs=-1",
+        "-mllvm:-gvn-hoist-max-depth=-1",
+        "-mllvm:-gvn-hoist-max-chain-length=-1",
+        "-mllvm:-enable-nontrivial-unswitch",
+        "-mllvm:-enable-ext-tsp-block-placement",
+        "-mllvm:-enable-constraint-elimination",
       ]
+
     } else {
       ldflags += [ "-flto=thin" ]
 
+      ldflags += [
+        "-Wl,-mllvm,-adce-remove-loops",
+        "-Wl,-mllvm,-enable-cse-in-irtranslator",
+        "-Wl,-mllvm,-enable-cse-in-legalizer",
+        "-Wl,-mllvm,-scalar-evolution-use-expensive-range-sharpening",
+        "-Wl,-mllvm,-loop-rotate-multi",
+        "-Wl,-mllvm,-enable-loopinterchange",
+        "-Wl,-mllvm,-enable-loop-distribute",
+        "-Wl,-mllvm,-enable-loop-flatten",
+        "-Wl,-mllvm,-enable-loop-versioning-licm",
+        "-Wl,-mllvm,-interleave-small-loop-scalar-reduction",
+        "-Wl,-mllvm,-enable-interleaved-mem-accesses",
+        "-Wl,-mllvm,-enable-masked-interleaved-mem-accesses",
+        "-Wl,-mllvm,-enable-gvn-hoist",
+        "-Wl,-mllvm,-gvn-hoist-max-bbs=-1",
+        "-Wl,-mllvm,-gvn-hoist-max-depth=-1",
+        "-Wl,-mllvm,-gvn-hoist-max-chain-length=-1",
+        "-Wl,-mllvm,-enable-nontrivial-unswitch",
+        "-Wl,-mllvm,-enable-ext-tsp-block-placement",
+        "-Wl,-mllvm,-enable-constraint-elimination",
+      ]
+
       # Enabling ThinLTO on Chrome OS too, in an effort to reduce the memory
       # usage in crbug.com/1038040. Note this will increase build time in
       # Chrome OS.
@@ -679,12 +746,12 @@ config("compiler") {
           "-Wcrl,object_path_lto" + _object_path_suffix,
         ]
       } else {
-        ldflags +=
-            [ "-Wl,--thinlto-cache-dir=" +
-              rebase_path("$root_out_dir/thinlto-cache", root_build_dir) ]
+        # ldflags +=
+        #     [ "-Wl,--thinlto-cache-dir=" +
+        #       rebase_path("$root_out_dir/thinlto-cache", root_build_dir) ]
       }
 
-      ldflags += [ "-Wl,--thinlto-cache-policy=$cache_policy" ]
+      # ldflags += [ "-Wl,--thinlto-cache-policy=$cache_policy" ]
 
       if (is_chromeos) {
         # ARM was originally set lower than x86 to keep the size
@@ -859,7 +926,7 @@ config("compiler") {
 # building with ThinLTO, no optimization is performed in the link step.
 config("thinlto_optimize_default") {
   if (!is_debug && use_thin_lto && is_a_target_toolchain) {
-    lto_opt_level = 0
+    lto_opt_level = 2
 
     if (is_win) {
       ldflags = [ "/opt:lldlto=" + lto_opt_level ]
@@ -882,7 +949,7 @@ config("thinlto_optimize_max") {
     if (thin_lto_enable_optimizations) {
       lto_opt_level = 2
     } else {
-      lto_opt_level = 0
+      lto_opt_level = 2
     }
 
     if (is_win) {
@@ -914,7 +981,8 @@ config("compiler_cpu_abi") {
     if (current_cpu == "x64") {
       cflags += [
         "-m64",
-        "-msse3",
+        # "-msse3",
+        "-mavx",
       ]
       ldflags += [ "-m64" ]
     } else if (current_cpu == "x86") {
@@ -2020,7 +2088,7 @@ if (is_win) {
     if ((!is_nacl || is_nacl_saigo) && is_clang) {
       # We don't care about unique section names, this makes object files a bit
       # smaller.
-      common_optimize_on_cflags += [ "-fno-unique-section-names" ]
+      # common_optimize_on_cflags += [ "-fno-unique-section-names" ]
     }
 
     common_optimize_on_ldflags += [
@@ -2074,7 +2142,7 @@ config("optimize") {
   } else if (optimize_for_size) {
     # Favor size over speed.
     if (is_clang) {
-      cflags = [ "-Oz" ] + common_optimize_on_cflags
+      cflags = [ "-O2" ] + common_optimize_on_cflags
 
       if (use_ml_inliner && is_a_target_toolchain) {
         cflags += [
@@ -2097,9 +2165,9 @@ config("optimize") {
     cflags = [ "-O2" ] + common_optimize_on_cflags
   }
   if (optimize_for_size) {
-    rustflags = [ "-Copt-level=s" ]
+    rustflags = [ "-Copt-level=s", "-Ctune-cpu=haswell", "-Ctarget-feature=+avx,+aes,+vaes", ]
   } else {
-    rustflags = [ "-Copt-level=3" ]
+    rustflags = [ "-Copt-level=3", "-Ctune-cpu=haswell", "-Ctarget-feature=+avx,+aes,+vaes", ]
   }
   ldflags = common_optimize_on_ldflags
 }
@@ -2165,9 +2233,9 @@ config("optimize_max") {
     } else if (optimize_for_fuzzing) {
       cflags = [ "-O1" ] + common_optimize_on_cflags
     } else {
-      cflags = [ "-O2" ] + common_optimize_on_cflags
+      cflags = [ "-O3" ] + common_optimize_on_cflags
     }
-    rustflags = [ "-Copt-level=3" ]
+    rustflags = [ "-Copt-level=3", "-Ctune-cpu=haswell", "-Ctarget-feature=+avx,+aes,+vaes", ]
   }
 }
 
@@ -2200,7 +2268,33 @@ config("optimize_speed") {
     } else {
       cflags = [ "-O3" ] + common_optimize_on_cflags
     }
-    rustflags = [ "-Copt-level=3" ]
+    rustflags = [ "-Copt-level=3", "-Ctune-cpu=haswell", "-Ctarget-feature=+avx,+aes,+vaes", ]
+  }
+}
+
+config("optimize_fast") {
+  if (is_nacl && is_nacl_irt) {
+    # The NaCl IRT is a special case and always wants its own config.
+    # Various components do:
+    #   if (!is_debug) {
+    #     configs -= [ "//build/config/compiler:default_optimization" ]
+    #     configs += [ "//build/config/compiler:optimize_max" ]
+    #   }
+    # So this config has to have the selection logic just like
+    # "default_optimization", below.
+    configs = [ "//build/config/nacl:irt_optimize" ]
+  } else {
+    ldflags = common_optimize_on_ldflags
+    if (is_win) {
+      # Favor speed over size, /O2 must be before the common flags.
+      # /O2 implies /Ot, /Oi, and /GF.
+      cflags = [ "/O2", "/clang:-ffast-math", ] + common_optimize_on_cflags
+    } else if (optimize_for_fuzzing) {
+      cflags = [ "-O1" ] + common_optimize_on_cflags
+    } else {
+      cflags = [ "-O3", "-ffast-math", ] + common_optimize_on_cflags
+    }
+    rustflags = [ "-Copt-level=3", "-Ctune-cpu=haswell", "-Ctarget-feature=+avx,+aes,+vaes", ]
   }
 }
 
@@ -2529,7 +2623,7 @@ config("minimal_symbols") {
 # names in the final debug information.
 config("no_symbols") {
   if (is_win) {
-    ldflags = [ "/DEBUG" ]
+    ldflags = [ "/DEBUG:NONE" ]
 
     # All configs using /DEBUG should include this:
     configs = [ ":win_pdbaltpath" ]
diff --git a/build/config/win/BUILD.gn b/build/config/win/BUILD.gn
index a5075393b1231..6a918cf197c6a 100644
--- a/build/config/win/BUILD.gn
+++ b/build/config/win/BUILD.gn
@@ -89,7 +89,7 @@ config("compiler") {
 
   cflags += [
     # Work around crbug.com/526851, bug in VS 2015 RTM compiler.
-    "/Zc:sizedDealloc-",
+    # "/Zc:sizedDealloc-",
   ]
 
   if (is_clang) {
@@ -130,7 +130,8 @@ config("compiler") {
     # microarchitecture. MSVC only supports a subset of architectures, and the
     # next step after SSE2 will be AVX.
     if (current_cpu == "x86" || current_cpu == "x64") {
-      cflags += [ "-msse3" ]
+      # cflags += [ "-msse3" ]
+     cflags += [ "/arch:AVX2", "-maes", "-mvaes", "/clang:-ffp-contract=fast" ]
     }
 
     if (exec_script("//build/win/use_ansi_codes.py", [], "trim string") ==
@@ -192,7 +193,7 @@ config("compiler") {
     if (use_lld) {
       # String tail merging leads to smaller binaries, but they don't compress
       # as well, leading to increased mini_installer size (crbug.com/838449).
-      ldflags += [ "/OPT:NOLLDTAILMERGE" ]
+      # ldflags += [ "/OPT:NOLLDTAILMERGE" ]
     }
 
     # TODO(siggi): Is this of any use anymore?
@@ -200,7 +201,7 @@ config("compiler") {
     # PDB file by about 5%) but does not otherwise alter the output binary. It
     # is enabled opportunistically for builds where it is not prohibited (not
     # supported when incrementally linking, or using /debug:fastlink).
-    ldflags += [ "/PROFILE" ]
+    # ldflags += [ "/PROFILE" ]
   }
 
   # arflags apply only to static_libraries. The normal linker configs are only
diff --git a/media/filters/ffmpeg_video_decoder.cc b/media/filters/ffmpeg_video_decoder.cc
index 78195a324a07b..0d012608862aa 100644
--- a/media/filters/ffmpeg_video_decoder.cc
+++ b/media/filters/ffmpeg_video_decoder.cc
@@ -40,7 +40,6 @@ static int GetFFmpegVideoDecoderThreadCount(const VideoDecoderConfig& config) {
     case VideoCodec::kUnknown:
     case VideoCodec::kVC1:
     case VideoCodec::kMPEG2:
-    case VideoCodec::kHEVC:
     case VideoCodec::kVP9:
     case VideoCodec::kAV1:
     case VideoCodec::kDolbyVision:
@@ -52,6 +51,7 @@ static int GetFFmpegVideoDecoderThreadCount(const VideoDecoderConfig& config) {
       // No extra threads for these codecs.
       break;
 
+    case VideoCodec::kHEVC:
     case VideoCodec::kH264:
     case VideoCodec::kVP8:
       // Normalize to three threads for 1080p content, then scale linearly
diff --git a/third_party/blink/renderer/core/BUILD.gn b/third_party/blink/renderer/core/BUILD.gn
index a42b76b65e9b0..01e478c99bb88 100644
--- a/third_party/blink/renderer/core/BUILD.gn
+++ b/third_party/blink/renderer/core/BUILD.gn
@@ -385,10 +385,10 @@ blink_core_sources("core_hot") {
   # TODO(ajwong) this is of mirror
   # https://source.chromium.org/chromium/chromium/src/+/master:third_party/blink/renderer/core/core.gni;l=26
   # to undo the configs which is fragile. Find a better way.
-  if (!is_debug && optimize_for_size) {
-    configs -= [ "//build/config/compiler:default_optimization" ]
-    configs += [ "//build/config/compiler:optimize_speed" ]
-  }
+  # if (!is_debug && optimize_for_size) {
+    # configs -= [ "//build/config/compiler:default_optimization" ]
+    configs += [ "//build/config/compiler:optimize_fast" ]
+  # }
 
   deps = [
     "//build:chromeos_buildflags",
