diff --git a/add-hevc-ffmpeg-decoder-parser.js b/add-hevc-ffmpeg-decoder-parser.js
index 663cdfa..83076fd 100644
--- a/add-hevc-ffmpeg-decoder-parser.js
+++ b/add-hevc-ffmpeg-decoder-parser.js
@@ -40,7 +40,6 @@ const patches = [
     ffmpeg_c_sources : [
       'libavcodec/autorename_libavcodec_bswapdsp.c',
       'libavcodec/dovi_rpu.c',
-      'libavcodec/dynamic_hdr10_plus.c',
       'libavcodec/dynamic_hdr_vivid.c',
       'libavcodec/hevc_cabac.c',
       'libavcodec/hevc_data.c',
@@ -254,4 +253,4 @@ function enableSoftwreDecodeHEVC() {
 }
 
 enableSoftwreDecodeHEVC();
-console.log('Modify ffmpeg success!');
\ No newline at end of file
+console.log('Modify ffmpeg success!');
